src/snowflake/cli/api/project/util.py

```diff
@@ -251,3 +248,23 @@ def identifier_to_show_like_pattern(identifier: str) -> str:
     matching this identifier
     """
     return f"'{escape_like_pattern(unquote_identifier(identifier))}'"
+
+
+def append_test_resource_suffix(identifier: str) -> str:
+    """
+    Append a suffix that should be added to specified account-level resources.
+
+    This is an internal concern that is currently only used in tests
+    to isolate concurrent runs and to add the test name to resources.
+    """
+    suffix = os.environ.get(TEST_RESOURCE_SUFFIX_VAR, "")
+    if identifier_to_str(identifier).endswith(identifier_to_str(suffix)):
+        # If the suffix has already been added, don't add it again
+        return to_identifier(identifier)
```

Comment:
This part was the issue I mentioned earlier. We shouldn't for it to_identifier here:
```
native_app:
  package:
    name: my.package
  application:
    name: <% ctx.native_app.package.name %>_app
```
should result in application name 'my.package_app', not '"my.package"_app'